### PR TITLE
added a git workflow for docker multiarch publish

### PR DESCRIPTION
**Pull request by krasi-georgiev**
closes: https://github.com/traccar/traccar-docker/issues/43

Only the alpine image is multiarch as the official java doesn't support multiarch for debian.

It will auto-publish for all branches and it is triggered by any push. I can also set it to be triggered by only changes to the dockerfiles, but that is error prone if you decide to refactor the folder structure at some point.

You can see an example of a successful push here
https://github.com/arribada/traccar-docker/actions/runs/181356434
https://hub.docker.com/repository/docker/arribada/traccar/tags?page=1

Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
ping @magna-z

**Comment by magna-z** [comment]
Why do you edit README.md?
Why do you use 4 spaces in YAML?

**Comment by krasi-georgiev** [comment]
> Why do you edit README.md?

because it is much easier to just mention that the images are constructed from the branch name instead of listing each one. less error prone.

> Why do you use 4 spaces in YAML?

not sure what happened, I fixed it.

**Comment by krasi-georgiev** [comment]
ping @magna-z

**Comment by toblie** [comment]
hi, I tried to deploy traccar/traccar:latest  and also traccar/traccar:4.10-alpine to my pine64 running armbian buster.
But badly both versions wasn't running and the log shows this error: 

standard_init_linux.go:211: exec user process caused "exec format error"

Can anyone help me to get it working? I think it also caused of arm compatibility? Is there anything to do to get your multiarch version running?

Thanks so much!
Tobi

**Comment by krasi-georgiev** [comment]
ping @magna-z

**Comment by magna-z** [comment]
**Files changed=598** - what is it???
This is small dedicated repository for build app in docker and nothing more!

**Comment by krasi-georgiev** [comment]
hhm , not sure what happened, that was a mistake, it is fixed now.

**Comment by krasi-georgiev** [comment]
ping @magna-z

**Comment by krasi-georgiev** [comment]
is the team interested in this contribution or shall I close it?

**Comment by tananaev** [comment]
I'll defer to @magna-z to decide.

**Comment by magna-z** [comment]
**README.md** is broken.
This is statically file with information about project. Why make a template out of it?

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  push-alpine:
+    runs-on: ubuntu-latest
+    name: Push alpine image
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@master
+      - name: Print Branch name
+        run: 'echo running on branch ${GITHUB_REF##*/}'
+      - name: Build and publish image
+        uses: ilteoood/docker_buildx@master
```
Comment (magna-z) [inline .github/workflows/docker.yaml:12]: Why not docker/build-push-action?

Comment (krasi-georgiev) [reply .github/workflows/docker.yaml:12]: yes I just found out about it as well. Will refactor.

**Comment by krasi-georgiev** [comment]
> This is statically file with information about project. Why make a template out of it?

I can leave as is, just thought that this way will reduce the manual work a bit. Can revert it if you prefer to keep these manual entries. Let me know

**Comment by tedenda** [comment]
I am also interesting to run Traccar in docker on RPi 4 8GB.
Any estimate on when this PR can be done and merged?

**Comment by krasi-georgiev** [comment]
This needs to be updated with the official multiarch docker build `docker/build-push-action` , but haven't had the time to refactor it. @tedenda feel free to take over this PR to refactor.

**Comment by tedenda** [comment]
I don't have the skills to do that I'm afraid. I'm a docker user that still only use prebuilt containers. Would like to learn to build custom made containers some day.

Hopefully someone else can help this PR over the finish line. Thanks for your efforts with this PR!

**Comment by tedenda** [comment]
As I said, this is not in my skillset, but maybe it is something like this?

```
    runs-on: ubuntu-latest
    name: Push alpine image
    steps:
      - 
        name: Checkout master
        uses: actions/checkout@master
      - 
        name: Print Branch name
        run: 'echo running on branch ${GITHUB_REF##*/}'
      - 
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
      -
        name: Login to DockerHub
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKER_USERNAME }}
          password: ${{ secrets.DOCKER_PASSWORD }}
      -
        name: Build and push
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./alpine/Dockerfile
          push: true
          tags: |
            '${GITHUB_REF##*/}-alpine'
            traccar/traccar:alpine
```

**Comment by tedenda** [comment]
> > This is statically file with information about project. Why make a template out of it?
> 
> I can leave as is, just thought that this way will reduce the manual work a bit. Can revert it if you prefer to keep these manual entries. Let me know

@magna-z Any answer to this question?

**Comment by magna-z** [comment]
Multi-platform docker images released for current version 4.11. [Details in README.md](https://github.com/traccar/traccar-docker#available-multi-platform-images)

**Pull request closed without merging**